src/config/configuration.validator.ts

```diff
@@ -0,0 +1,23 @@
+import Ajv, { Schema } from 'ajv';
+
+const configurationSchema: Schema = {
+  type: 'object',
+  properties: {
+    AUTH_TOKEN: { type: 'string' },
+    EXCHANGE_API_KEY: { type: 'string' },
+  },
+  required: ['AUTH_TOKEN', 'EXCHANGE_API_KEY'],
+};
+
+export function validate(
+  configuration: Record<string, unknown>,
+): Record<string, unknown> {
+  if (process.env.NODE_ENV !== 'test') {
+    const ajv = new Ajv({ allErrors: true });
+    if (!ajv.validate(configurationSchema, configuration)) {
+      const errors = ajv.errors?.reduce((acc, e) => [...acc, e.message], []);
```

Comment:
This is using the reduce function but the accumulator is an array that is being built with the spread operator. Why not have the initial value as an empty array and add the error message to it?

Reply:
> Why not have the initial value as an empty array and add the error message to it?

Yes, that was the purpose of the reduce function, but in this case I think the code can be simplified by just using a map function. I've changed it in [56eed68](https://github.com/safe-global/safe-client-gateway-nest/pull/307/commits/56eed689b2b7473d50ecf3b5f2dec8db6e3d0463), I hope now the purpose is clearer.